test/libsolidity/syntaxTests/functionTypes/warn_deprecate_gas_function.sol

```diff
@@ -0,0 +1,8 @@
+contract C {
+    function (uint) external payable returns (uint) x;
+    function f() public {
+        x.gas(2)(1);
```

Comment:
This now only checks "function type state variables". Can you add a proper external function "y" to this and check ``y.gas(...)`` on it as well?
Also we should have tests for low-level calls, so e.g. ``a.call.gas(2)`` for an ``address a``.
Similarly for ``value``.
So generally, it would be good to have one example of every kind of object that has a "gas" member here.